src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Tests;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Http.Functional.Tests
+{
+    // Note:  Disposing the HttpClient object automatically disposes the handler within. So, it is not necessary
+    // to separately Dispose (or have a 'using' statement) for the handler.
+    public class PostScenarioTest
+    {
+        private const string ExpectedContent = "Test contest";
+        private const string UserName = "user1";
+        private const string Password = "password1";
+        private readonly static Uri BasicAuthServerUri =
+            HttpTestServers2.BasicAuthUriForCreds(false, UserName, Password);
+        private readonly static Uri SecureBasicAuthServerUri =
+            HttpTestServers2.BasicAuthUriForCreds(true, UserName, Password);
+
+        readonly ITestOutputHelper _output;
+
+        public static object[][] EchoServers
+        {
+            get
+            {
+                return HttpTestServers2.EchoServers;
+            }
+        }
+
+        public static object[][] BasicAuthEchoServers
+        {
+            get
+            {
+                return new object[][]
+                    {
+                        new object[] { BasicAuthServerUri },
+                        new object[] { SecureBasicAuthServerUri }
+                    };
+            }
+        }
+
+        public PostScenarioTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Theory, MemberData("EchoServers")]
+        public async Task PostUsingContentLengthSemantics_Success(Uri serverUri)
+        {
+            await PostHelper(serverUri, ExpectedContent, new StringContent(ExpectedContent), true, false);
```

Comment:
Consider naming the Boolean arguments to make it easier to understand what's being configured, e.g.
```
await PostHelper(serverUri, ExpectedContent, new StringContent(ExpectedContent), 
    useContentLengthUpload: true, useChunkedEncodingUpload: false);
```